lib/ansiblelint/errors.py

```diff
@@ -12,10 +15,17 @@ def __init__(self, message=None, linenumber=0, line=None, filename=None, rule=No
         """Initialize a MatchError instance."""
         super().__init__(message)
 
+        if not (message or rule):
+            self.linenumber = 0
```

Comment:
@ssbarnea could please explain why this is in the branch that will explode anyway?

Reply:
Should it be just removed? (judging from the logic I see)
```suggestion
```